File: problemas/ad-hoc/3024/3024.swift
import Foundation

let entrada = readLine(strippingNewline: true)!.components(separatedBy: " ")
let n = Int(entrada[0])!
let x = Int(entrada[1])!

let alturas = readLine(strippingNewline: true)!.components(separatedBy: " ")

var resposta = 1
var acumulado = 1
for i in 1..<n {
    if (Int(alturas[i])! - Int(alturas[i - 1])! <= x) {
        acumulado += 1
    } else {
        resposta = max(resposta, acumulado)
        acumulado = 1
    }
}
resposta = max(resposta, acumulado)

print(resposta)

File: problemas/ad-hoc/3024/3024.r
input <- file('stdin', 'r')

entrada <- as.integer(strsplit(readLines(input, n = 1), ' ')[[1]])
n <- entrada[1]
x <- entrada[2]

alturas <- as.integer(strsplit(readLines(input, n = 1), ' ')[[1]])

resposta <- 1
acumulado <- 1

if(n > 1) {
    for(i in 2:n) {
        if (alturas[i] - alturas[i - 1] <= x) {
            acumulado <- acumulado + 1
        } else {
            resposta <- max(resposta, acumulado)
            acumulado <- 1
        }
    }
    resposta <- max(resposta, acumulado)
}

write(paste(resposta), '')
